src/backend/PortabilityService.WorkflowManagement/WorkflowManager.cs

```diff
@@ -84,7 +81,7 @@ public static WorkflowQueueMessage GetFirstStage(string submissionId)
         public async Task<WorkflowQueueMessage> ExecuteActionsToNextStage(WorkflowQueueMessage currentMsg, CancellationToken cancelToken)
         {
             //Get action corresponding to the current message's workflow stage
-            Debug.Assert(actions.Length - 1 > (int)currentMsg.Stage, "Stage must be within bounds of Actions array.");
+            Debug.Assert((int)currentMsg.Stage < actions.Length, "Stage must be within bounds of Actions array.");
```

Comment:
Should we have this throw instead of be debug? That way we can catch it in production if it ever occurs

Reply:
I think it makes sense as a debug assert because `currentMsg.Stage` is an enum. If the assertion fails, that's a bug. I'm also thinking we won't need, and so won't deploy, this code.